framework/src/controller/migrations/migration.js

```diff
@@ -253,31 +274,82 @@ class Migration extends BaseEntity {
 						migration && {
 							id: migration[1],
 							name: migration[2],
-							path: path.join('../sql/migrations/updates', migrationFile),
+							path: path.join(updatesPath, migrationFile),
 						}
 					);
 				})
-				.sort((a, b) => a.id - b.id) // Sort by migration ID, ascending
 				.filter(
 					migration =>
 						migration &&
-						fs
-							.statSync(path.join(this.sqlDirectory, migration.path))
-							.isFile() &&
+						fs.statSync(migration.path).isFile() &&
 						(!lastMigrationId || +migration.id > lastMigrationId)
 				)
-				.map(f => {
-					f.file = this.adapter.loadSQLFile(f.path, this.sqlDirectory);
-					return f;
+				.sort((a, b) => a.id - b.id) // Sort by migration ID, ascending
+				.map(migration => {
+					migration.file = this.adapter.loadSQLFile(migration.path, '');
+					return migration;
+				})
+		);
+	}
```

Comment:
Its not a good approach to maintain two migrations tables while there is no clear physical segregation. Every migration file runs on same database and modify schema of same database. 

You can use the same migrations table to run framework migrations with the namespace `framework`.

Reply:
I did the separation because I couldn't find a way to migrate the `migrations` table using the same logic.

Reply:
`migrations` table not required any migration. Its a pre-requisite come along the framework. If you want to introduce some change in the migrations table for future you still can register any SQL file with the namespace `lisk` or `lisk-framework` that will any case should and must be executed in the timely order. So no need of this new concept internal migrations.

Reply:
Current `migrations` table has no `namespace` column. Hence, it requires migration. And when the table changes its schema itself you can't use the same code/logic for before and after the migration is run.

Reply:
Each migration run in a consolidated database transaction. So you can change structure of migrations table within the migration file. Then left would be when adding entry to migrations table. That you can have two versions to maintain backward compatibility for old migrations.

Reply:
I think solution that @lsilvs have is a good solution unless if you have alternative.
He said, migration file to alter migrations table is possible, but code cannot be adopted.
1.6 => 2.0 works, but fresh starting database will not work in this case.